core/server/metrics/format-converters.js

```diff
@@ -1,11 +1,14 @@
-import groupBy from 'lodash.groupby'
-
 function promClientJsonToInfluxV2(metrics, extraLabels = {}) {
   return metrics
     .flatMap(metric => {
-      const valuesByLabels = groupBy(metric.values, value =>
-        JSON.stringify(Object.entries(value.labels).sort()),
-      )
+      const valuesByLabels = metric.values.reduce((acc, value) => {
```

Comment:
This implementation seems correct, but I did have something in mind as this runs often.
I looked the `_.groupBy` implementation and it seems to have [an optimization](https://github.com/lodash/lodash/blob/5c7ff7a9f3aae60e440dfb8b11a8a873e38cb1d2/lodash.groupby/index.js#L783) for large arrays, i think they use map instead of an array.
I'm not sure but doesn't we send a large array here as we seem to send every [1000 seconds the data](https://github.com/badges/shields/blob/959e1c8bec349e8c2eff65302543a7ddbdd128b6/core/server/influx-metrics.js#L47)? Should we consider optimization as well or do you think this is not a hotspot?

Reply:
> we seem to send every 1000 seconds the data

We send metrics every 15 seconds (unless we've overridden the default, but I don't think we have): https://github.com/badges/shields/blob/959e1c8bec349e8c2eff65302543a7ddbdd128b6/config/default.yml#L11

There are mixed claims regarding performance of lodash groupBy vs. reduce on the Internet. But ultimately yeah, grouping metrics every now and again will not be a hotspot, so I don't think we should worry about micro-optimisation too much here :)